samcli/commands/_utils/custom_options/hook_name_option.py

```diff
@@ -215,3 +221,16 @@ def _read_parameter_value(param_name, opts, ctx, default_value=None):
     Read SAM CLI parameter value either from the parameters list or from the samconfig values
     """
     return opts.get(param_name, ctx.default_map.get(param_name, default_value))
+
+
+def record_hook_telemetry(opts: Mapping[str, Any]):
+    """
+    Emit metrics related to hooks based on the options passed into the command
+
+    Parameters
+    ----------
+    opts: Mapping[str, Any]
+        Mapping between a command line option and its value
+    """
+    if PLAN_FILE_OPTION in opts and opts.get(PLAN_FILE_OPTION) is not None:
```

Comment:
Does this also check for if the plan file option was passed in via config file?

Reply:
Good call, it did not. Addressed this in the newest commit.